### ISSUE TITLE
Déclinaison Rmarkdown

### ISSUE THREAD

**Issue by tvroylandt** [spyrales/gouvdown#4]
Faire 4 CSS pour RMarkdown :

+ un global assez light ;
+ un pour les fonts ;
+ un pour `pagedown` ;
+ un pour `xaringan`.

Normalement ça devrait combler les besoins.

**Comment by MaelTheuliere** [comment]
On peut rajouter pkgdown, bookdown, rolldown
Je pense que cette issue est à scinder en autant d'output

**Comment by tvroylandt** [comment]
Oui certes. Mais normalement le global devrait permettre de faire la majorité des exports et le reste n'est que de l'ajustement par output.

J'utilise vraiment que `pagedown`, `xaringan` et `bookdown`  donc je peux pas dire pour les autres. On va commencer par là et on ajustera ensuite.

**Comment by MaelTheuliere** [comment]
ok

**Comment by RLesur** [comment]
A propos de `bookdown` : vous parlez de quel _output format_ précisément ? `bookdown::gitbook()`,  `bookdown::html_document2()`, `bookdown::odt_document2()`... ? Je comprends implicitement `bookdown::gitbook()` mais je n'en suis pas complètement certain.

En plus des feuilles de style CSS, on pourrait peut-être récapituler les _output formats_ qu'on souhaite avoir et surtout leur donner un nom (c'est certainement le plus difficile).

Voici un premier essai de liste d'_output formats_ qu'on aimerait avoir ainsi que des propositions de noms de fonctions : 

- `bookdown::html_document2()` -> `gouvdown::html_gouv()`
- `bookdown::gitbook()` -> `gouvdown::gitbook_gouv()`
- `xaringan::moon_reader()` -> `gouvdown::diapo_gouv()`
- `pagedown::html_paged()` -> `gouvdown::html_paged_gouv()`
- `rolldown::scrollama()` -> `gouvdown::scroll_gouv()`

Pour `pkgdown`, il n'y a pas d'output format, tout est à mettre dans `inst/pkgdown` : cependant, je me demande si sa place est bien dans `gouvdown` ou si un package spécifique doit être fait pour un thème `pkgdown` spécifique (c'est ce qui est fait ailleurs). D'ailleurs, je ne vois pas bien à ce stade comment gérer les polices dans un thème `pkgdown` open source :thinking:

**Comment by tvroylandt** [comment]
Et dans l'idéal un format note simple aussi, type ce qui se fait dans `memor`. Ca peut être à partir de `pagedown`, avec une version qui a une page de garde, toc, etc... et une présentation plus "dossier" et une autre qui est plus "light"

**Comment by RLesur** [comment]
> Et dans l'idéal un format note simple aussi, type ce qui se fait dans `memor`. Ca peut être à partir de `pagedown`, avec une version qui a une page de garde, toc, etc... et une présentation plus "dossier" et une autre qui est plus "light"

Bien vu. Il faudrait peut-être 2 output formats différents dans ce cas ?

**Comment by tvroylandt** [comment]
Oui, ça peut être `gouvdown::html_paged_note_gouv()` et `gouvdown::html_paged_dossier_gouv()`

**Comment by RLesur** [comment]
> Oui, ça peut être `gouvdown::html_paged_note_gouv()` et `gouvdown::html_paged_dossier_gouv()`

on peut peut-être laisser tomber `paged` dans ce cas, et si on francise : pourquoi pas `gouvdown::note_html()` et `gouvdown::dossier_html()`. J'avais rajouté le suffixe `_gouv` pour éviter les conflits de nom de fonction mais là c'est peut-être pas nécessaire de le rajouter :thinking:

**Comment by tvroylandt** [comment]
On peut enlever html aussi... parce que le but de `pagedown` (en tout cas pour nous ici) c'est de sortir du PDF. Donc ça peut prêter à confusion aux moins informés qui voudront juste l'utiliser.

Et pareil je collerai bien un knit PDF en default dedans si possible.

**Comment by RLesur** [comment]
> On peut enlever html aussi... parce que le but de `pagedown` (en tout cas pour nous ici) c'est de sortir du PDF. Donc ça peut prêter à confusion aux moins informés qui voudront juste l'utiliser.
> 
> Et pareil je collerai bien un knit PDF en default dedans si possible.

J'avais laissé le suffixe `_html` au cas où quelqu'un aurait voulu faire un `note_odt` par exemple.

Mais je vois, tu veux te débarrasser de LaTeX, MSOffice et LibreOffice encore plus vite que moi :smile:  Ca me va bien d'avoir simplement `note()` et `dossier()` quitte à faire des variantes ensuite.

**Comment by tvroylandt** [comment]
All hail CSS !

**Comment by tvroylandt** [comment]
Désolé je suis à la bourre totale sur le côté Rmd. J'ai eu pas mal de choses en ce moment à gérer.
J'essaye de m'y mettre bientôt !

D'ici là on peut faire un peu de teasing en affichant des graphs avec le thème et les palettes ?

**Comment by MaelTheuliere** [comment]
Complètement !

**Comment by tvroylandt** [comment]
J'ai fait une issue à part https://github.com/spyrales/gouvdown/issues/17

**Comment by RLesur** [comment]
> Désolé je suis à la bourre totale sur le côté Rmd. J'ai eu pas mal de choses en ce moment à gérer.
> J'essaye de m'y mettre bientôt !

Pareil de mon côté :disappointed:
